src/mlpack/methods/ann/layer/convolution.hpp

```diff
@@ -162,6 +162,45 @@ class Convolution
   //! Modify the output height.
   size_t& OutputHeight() { return outputHeight; }
 
+  //! Get the input size
+  size_t const& InputSize() const { return inSize; }
+
+  //! Get the output size.
+  size_t const& OutputSize() const { return outSize; }
+
+  //! Get the kernel dimensions.
+  std::tuple<size_t&, size_t&> const& KernelDims() const
+  {
+    return std::forward_as_tuple(kH, kW);
+  }
+  //! Modify the kernel dimensions.
+  std::tuple<size_t&, size_t&> KernelDims()
+  {
+    return std::forward_as_tuple(kH, kW);
+  }
+
+  //! Get the stride dimensions.
+  std::tuple<size_t&, size_t&> const& StrideDims() const
+  {
+    return std::forward_as_tuple(dH, dW);
+  }
+  //! Modify the stride dimensions.
+  std::tuple<size_t&, size_t&> StrideDims()
+  {
+    return std::forward_as_tuple(dH, dW);
+  }
+
+  //! Get the padding dimensions.
```

Comment:
Would a return type of ```std::tuple<size_t, size_t>``` instead of ```std::tuple<size_t&, size_t&>``` be better for the getter functions (only rvalue return type ones, not the ones that allow modification)?

Reply:
We use `const` to prevent modification. So maybe you could just remove the function without `const` if you don't want to allow modification.

Reply:
using ``std::tuple<size_t, size_t>`` might be more intuitive, but @akhandait is right const, will prevent modification anyway, personally I would probably go with ``std::tuple<size_t, size_t>``.